src/components/ModalCentered.tsx

```diff
@@ -0,0 +1,176 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { motion } from 'framer-motion'
+import { X } from 'lucide-react'
+import { FC, ReactNode } from 'react'
+import styled, { css } from 'styled-components'
+
+import Button from './Button'
+import Modal, { ModalBackdrop, ModalProps } from './Modal'
+import { Section } from './PageComponents/PageContainers'
+import PanelTitle, { TitleContainer } from './PageComponents/PanelTitle'
+import Spinner from './Spinner'
+
+interface ModalCenteredProps extends ModalProps {
+  title?: string
+  subtitle?: string
+  isLoading?: boolean
+  header?: ReactNode
+}
+
+const ModalCentered: FC<ModalCenteredProps> = ({
+  title,
+  subtitle,
+  onClose,
+  focusMode,
+  isLoading,
+  header,
+  children
+}) => (
+  <Modal onClose={onClose} focusMode={focusMode}>
+    <StyledModal
```

Comment:
Are we wrapping successively 2 animated components? (Modal, and StyledModal?).
We could maybe simply allow overriding the `motion` related props in the `Modal` component? (initial, animate, exit and transition).

Something like:

```ts
export interface ModalProps extends ComponentPropsWithoutRef<typeof motion.div> {
  onClose: () => void
  focusMode?: boolean
  hasPadding?: boolean
}

const Modal: FC<ModalProps> = ({
  children,
  onClose,
  focusMode,
  initial = { opacity: 0 },
  animate = { opacity: 1 },
  exit = { opacity: 0 },
  transition = { duration: 0.2, ease: 'easeOut' },
  className,
  hasPadding = true
}) => {

  [...]

  return (
    <motion.div initial={initial} animate={animate} exit={exit} transition={transition} className={className}>
      {children}
      <ModalBackdrop onClick={onClose} focusMode={focusMode} />
    </motion.div>
  )
}

export default styled(Modal)`
  position: fixed;
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
  display: flex;
  padding: ${({ hasPadding }) => hasPadding && css`var(--spacing-4)`};
  z-index: 1000;
`
```

Reply:
Hm, but do we rly want to override the animation? We want the overall backdrop to fade in (not move vertically) and the actual modal to fade in AND move vertically.

But I agree that maybe it's not the best to have 2 successive animated components. That's why I propose to move the first animation only on the backdrop, and leave the second animation where it belongs. See diff:

<details><summary>Click to see diff</summary>

```diff
diff --git a/src/components/Modal.tsx b/src/components/Modal.tsx
index 9660639..55c37aa 100644
--- a/src/components/Modal.tsx
+++ b/src/components/Modal.tsx
@@ -18,15 +18,16 @@ along with the library. If not, see <http://www.gnu.org/licenses/>.
 
 import { motion } from 'framer-motion'
 import { FC, useCallback, useEffect } from 'react'
-import styled, { css } from 'styled-components'
+import styled from 'styled-components'
 
 export interface ModalProps {
   onClose: () => void
   focusMode?: boolean
   hasPadding?: boolean
+  className?: string
 }
 
-const Modal: FC<ModalProps> = ({ children, onClose, focusMode, hasPadding = true }) => {
+const Modal: FC<ModalProps> = ({ children, onClose, focusMode, hasPadding = true, className }) => {
   // Prevent body scroll on mount
   useEffect(() => {
     document.body.style.overflow = 'hidden'
@@ -54,31 +55,32 @@ const Modal: FC<ModalProps> = ({ children, onClose, focusMode, hasPadding = true
   }, [handleEscapeKeyPress, onClose])
 
   return (
-    <ModalContainer
-      initial={{ opacity: 0 }}
-      animate={{ opacity: 1 }}
-      exit={{ opacity: 0 }}
-      transition={{ duration: 0.2, ease: 'easeOut' }}
-      hasPadding={hasPadding}
-    >
+    <div className={className}>
       {children}
-      <ModalBackdrop onClick={onClose} focusMode={focusMode} />
-    </ModalContainer>
+      <ModalBackdrop
+        initial={{ opacity: 0 }}
+        animate={{ opacity: 1 }}
+        exit={{ opacity: 0 }}
+        transition={{ duration: 0.2, ease: 'easeOut' }}
+        onClick={onClose}
+        focusMode={focusMode}
+      />
+    </div>
   )
 }
 
-const ModalContainer = styled(motion.div)<{ hasPadding?: boolean }>`
+export default styled(Modal)<{ hasPadding?: boolean }>`
   position: fixed;
   top: 0;
   bottom: 0;
   right: 0;
   left: 0;
   display: flex;
-  padding: ${({ hasPadding }) => hasPadding && css`var(--spacing-4)`};
+  padding: ${({ hasPadding }) => hasPadding && 'var(--spacing-4)'};
   z-index: 1000;
 `
 
-export const ModalBackdrop = styled.div<{ focusMode?: boolean; light?: boolean }>`
+export const ModalBackdrop = styled(motion.div)<{ focusMode?: boolean; light?: boolean }>`
   position: absolute;
   top: 0;
   right: 0;
@@ -95,5 +97,3 @@ export const ModalBackdrop = styled.div<{ focusMode?: boolean; light?: boolean }
       ? 'rgba(0, 0, 0, 0.15)'
       : 'rgba(0, 0, 0, 0.6)'};
 `
-
-export default Modal
diff --git a/src/components/ModalCentered.tsx b/src/components/ModalCentered.tsx
index d40a0ed..04bfa71 100644
--- a/src/components/ModalCentered.tsx
+++ b/src/components/ModalCentered.tsx
@@ -44,7 +44,7 @@ const ModalCentered: FC<ModalCenteredProps> = ({
   children
 }) => (
   <Modal onClose={onClose} focusMode={focusMode}>
-    <StyledModal
+    <CenteredBox
       initial={{ opacity: 0, y: 20 }}
       animate={{ opacity: 1, y: 0 }}
       exit={{ opacity: 0, y: 20 }}
@@ -71,7 +71,7 @@ const ModalCentered: FC<ModalCenteredProps> = ({
           </ModalLoadingSpinner>
         </>
       )}
-    </StyledModal>
+    </CenteredBox>
   </Modal>
 )
 
@@ -88,7 +88,7 @@ export const HeaderLogo = styled.div`
   width: 100%;
 `
 
-const StyledModal = styled(motion.div)`
+const CenteredBox = styled(motion.div)`
   display: flex;
   flex-direction: column;
   margin: auto;
```

</details>

Reply:
Yes, LGTM.

Reply:
done